src/inmanta/data/__init__.py

```diff
@@ -2151,19 +2175,22 @@ async def delete_cascade(self) -> None:
                 # Delete all code associated with this version
                 await Code.delete_all(connection=con, environment=self.environment, version=self.version)
 
-                # Delete version and all resources (FK)
```

Comment:
Also document the locking order in one of the classes, and have the docstring of the others refer to that one, such that a developer working one any of these classes has a high probability of finding this information while preventing replication.